### PR TITLE
Share

### PR DESCRIPTION
**Pull request by ksheremet**

### CODE REVIEW

File: lib/widgets/send_invite.dart [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import 'package:flutter/material.dart';
+import 'package:share/share.dart';
+
+import '../flutter/localization.dart';
+
+sendInvite(BuildContext context) {
```
Comment (dotdoom) [inline lib/widgets/send_invite.dart]: `void`

File: lib/pages/deck_sharing.dart [hunk 2]
```diff
@@ -97,6 +96,21 @@ class _DeckSharingState extends State<DeckSharingPage> {
       reportError("Deck sharing exception:", e, stackTrace);
     }
   }
+
+  _inviteUser() async {
```
Comment (dotdoom) [inline lib/pages/deck_sharing.dart]: `void`

File: pubspec.yaml [hunk 3]
```diff
@@ -28,6 +28,8 @@ dependencies:
 
   http: "^0.11.3+16"
 
+  share: "^0.5.2"
```
Comment (dotdoom) [inline pubspec.yaml]: consistency: remove quotes (also above)

File: lib/flutter/localization.dart [hunk 4]
```diff
@@ -258,6 +258,38 @@ class AppLocalizations {
...
+      name: 'appNotInstalledSharingDeck',
+      desc: 'The app hasn\'t installed by user with who deck was shared',
+    );
+  }
+
+  String get inviteToAppMessage {
+    return Intl.message(
+      '''I invite you to install Delern, a spaced repetition learning app, which will allow you to learn quickly and easily!
+  
+  Proceed to https://play.google.com/store/apps/details?id=org.dasfoo.delern to install it from Google Play!
```
Comment (dotdoom) [inline lib/flutter/localization.dart]: I think the spaces in the beginning of these lines are redundant and harmful for the reader of the invitation.

WDYT?

File: lib/l10n/intl_ru.arb [hunk 5]
```diff
@@ -202,5 +202,26 @@
...
+    }
+  },
+  "appNotInstalledSharingDeck": "Данный пользователь еще не установил Delern. Отправить приглашение?",
+  "@appNotInstalledSharingDeck": {
+    "description": "The app hasn't installed by user with who deck was shared",
+    "type": "text",
+    "placeholders": {
+    }
+  },
+  "inviteToAppMessage": "Я приглашаю Вас установить Delern, система интервального изучения, которая позволяет изучать быстро и легко!\n  \n  Пройдите по ссылке https://play.google.com/store/apps/details?id=org.dasfoo.delern чтобы установить из Google Play!\n  \n  После установки, следите за последними новостями Delern:\n  Facebook: https://fb.me/das.delern\n  VKontakte: https://vk.com/delern\n  Google+: https://plus.google.com/communities/104603840044649051798\n  Twitter: https://twitter.com/dasdelern",
```
Comment (dotdoom) [inline lib/l10n/intl_ru.arb]: систем**у** ?

File: lib/pages/deck_sharing.dart [hunk 6]
```diff
@@ -97,6 +96,21 @@ class _DeckSharingState extends State<DeckSharingPage> {
...
+    var locale = AppLocalizations.of(context);
+    var inviteUser = await showSaveUpdatesDialog(
+        context: context,
+        changesQuestion: locale.appNotInstalledSharingDeck,
+        yesAnswer: locale.send,
+        noAnswer: locale.cancel);
+    if (inviteUser) {
+      sendInvite(context);
+      setState(() {
```
Comment (dotdoom) [inline lib/pages/deck_sharing.dart]: I'd suggest to `setState` *outside* this method - for isolation.

File: lib/pages/deck_sharing.dart (hunk 2 above)
Comment (ksheremet) [reply lib/pages/deck_sharing.dart]: Future<void>

**Pull request merged**